curriculum/challenges/english/25-front-end-development/review-react-basics/67487e141bb6a7140a352e12.md

```diff
@@ -0,0 +1,253 @@
+- Frameworks, on the other hand, provide a more defined structure for building applications. They often come with a set of rules and conventions that developers need to follow.
+- Examples of frameworks include Angular and React.
+- **Single page applications**(SPAs) are web applications that load a single HTML page and dynamically update that page as the user interacts with the application without reloading the entire page.
+- SPAs use JavaScript to manage the application's state and render content. This is often done using frameworks like React, Vue, or Angular, which provide great tools for building complex user interfaces.
+- Some issues surrounding SPAs include:
+
+  - Screenreaders struggle struggle with dynamicaaly updated content.
+  - The URL does not change when the user navigates within the application, which can make it difficult to bookmark, backtrack or share specific pages.
+  - Initial load time can be slow if the application is large as all the assets need to be loaded upfront.
+
+## React- a JavaScript library for building user interfaces
+
+- React is a popular JavaScript library for building user interfaces and web applications.
+- A core concept of React is creation of reusable UI components that can update and render independently as data changes.
+- React allows developers to describe how the UI should look like based on the application state. React then updates and renders the right components when the data or the state changes.
+
+## React Components
+
+- Components are the building blocks of React applications that allow developers to break down complex user interfaces into smaller, manageable pieces.
+- The UI is described using JSX, a syntax extension that allows developers to write HTML-like code within JavaScript.
+- Components are basically JS functions or classes that return a piece of UI. 
+Here is an example of a simple React component that renders a greeting message:
+
+```jsx
+function Greeting() {
+    const name = 'Anna';
+    return <h1>Welcome, {name}!</h1>;
+}
+```
+
+To use the component, you can simply call:
+    
+```jsx
+    <Greeting />
+```
+
+## Meta frameworks
+
+- Meta frameworks, also known as full-stack frameworks, are tools that are build upon existing libraries or frameworks to provide additional features like server-side rendering (SSR), static site generation (SSG), routing, code splitting, and optimized build processes out of the box.
+- In the React ecosystem, Next.js is an example of a meta framework.
+
+Here is an example of a simple Next.js page that renders the same greeting message as the previous example:
+
+```jsx
+function Greeting() {
+    const name = 'Anna';
+    return <h1>Welcome, {name}!</h1>;
+}
+
+export default Greeting
+```
+
+- Meta frameworks can perform server-side rendering. This means that the initial HTML is rendered on the server, which can lead to faster page loads and improved SEO.
+
+## CLI tools for project setup
```

Comment:
This section also needs to be updated to match the script.
The script now focus just on how to setup a react app using vite

Reply:
updated